File: Java-Spring/src/test/java/org/codecop/lcdnumbers/Input.java
package org.codecop.lcdnumbers;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Shortcut methods to get input test data.
 */
public class Input {

    public static List<String> allDigitsSize1() throws IOException {
        return read("numbers grid 3x5 - size 1.txt");
    }

    public static List<String> allDigitsSize2() throws IOException {
        return read("numbers grid 4x7 - size 2.txt");
    }

    public static List<String> allDigitsSize3() throws IOException {
        return read("numbers grid 5x9 - size 3.txt");
    }

    private static List<String> read(String fileName) throws IOException {
        return Files.readAllLines(Paths.get("src/test/resources", fileName));
    }

    public static String join(List<String> lines) {
        return lines.stream().collect(Collectors.joining("\n")) + "\n";
    }

}

File: Kotlin-Spring/src/test/kotlin/org/codecop/lcdnumbers/Input.kt
package org.codecop.lcdnumbers

/**
 * Shortcut methods to get input test data.
 */

import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths
import java.util.stream.Collectors


fun allDigitsSize1(): List<String> {
    return read("numbers grid 3x5 - size 1.txt")
}

fun allDigitsSize2(): List<String> {
    return read("numbers grid 4x7 - size 2.txt")
}

fun allDigitsSize3(): List<String> {
    return read("numbers grid 5x9 - size 3.txt")
}

private fun read(fileName: String): List<String> {
    return Files.readAllLines(Paths.get("src/test/resources", fileName))
}

fun List<String>.joinLines() = joinToString("\n", postfix = "\n")